crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -368,30 +368,52 @@ impl<T: TransactionOrdering> TxPool<T> {
                         self.all_transactions.pending_fees.base_fee,
                     ))
                 } else {
-                    Box::new(self.pending_pool.best())
+                    // Blob fee may increase.
+                    Box::new(self.pending_pool.best_with_basefee_and_blobfee(
+                        best_transactions_attributes.basefee,
+                        best_transactions_attributes.blob_fee.unwrap_or_default(),
+                    ))
```

Comment:
we already check the blob fee above in L359

can we convert this into another match statement so that we can cover all branches